cwlavro-tools/src/main/java/io/cwl/avro/CWL.java

```diff
@@ -103,6 +105,48 @@ private String convertCWLType(Object cwlType) {
         return type;
     }
 
+    private static Object getStubForItem(String strType, Object stub, String value) {
+        final Map<String, String> file = new HashMap<>();
+
+        switch (strType) {
+            case "File":
+                file.put("class", "File");
+                file.put("path", value != null ? value : "/tmp/fill_me_in.txt");
+                stub = file;
+                break;
+            case "Directory":
+                file.put("class", "Directory");
+                file.put("path", value != null ? value : "/tmp/fill_directory_in");
+                stub = file;
+                break;
+            case "boolean":
+                stub = value != null? Boolean.parseBoolean(value) : Boolean.FALSE;
+                break;
+            case "int":
+                stub = value != null? Integer.parseInt(value) :0;
+                break;
+            case "long":
+                stub = value != null? Long.parseLong(value) :0L;
+                break;
+            case "float":
+                stub = value != null? Float.parseFloat(value) :0.0;
+                break;
+            case "double":
+                stub = value != null? Double.parseDouble(value) : Double.MAX_VALUE;
+                break;
+            default:
+                break;
+        }
+        return stub;
+    }
+
+    private static Object getStubForArray(String strType, Object stub, String value) {
+        final List<Object> list = new ArrayList<>();
+        stub = getStubForItem(strType, stub, value);
```

Comment:
Avoid re-assigning input parameters, it can get confusing 
https://softwareengineering.stackexchange.com/questions/245767/is-modifying-an-incoming-parameter-an-antipattern